nextstrain/cli/command/check_setup.py

```diff
@@ -103,16 +106,26 @@ def run(opts: Options) -> int:
 
     if supported_runners:
         print("Supported Nextstrain environments:", ", ".join(success(runner_name(r)) for r in supported_runners))
-
-        if opts.set_default:
-            default_runner = supported_runners[0]
-            print()
-            print("Setting default environment to %s." % runner_name(default_runner))
-            config.set("core", "runner", runner_name(default_runner))
-            default_runner.set_default_config()
     else:
         print(failure("No support for any Nextstrain environment."))
 
+    # Set default if requested.
+    set_default = None
+
+    if opts.set_default is True:
+        if supported_runners:
+            set_default = supported_runners[0]
+    elif opts.set_default:
+        set_default = all_runners_by_name[opts.set_default]
+
+    if set_default:
+        default_runner = set_default
+        print()
+        print("Setting default environment to %s." % runner_name(default_runner))
+        config.set("core", "runner", runner_name(default_runner))
+        default_runner.set_default_config()
```

Comment:
Output is a bit confusing in this scenario:

```sh
# Docker daemon is not running
nextstrain check-setup --set-default docker
```

Output:

```
# docker is not supported

...

Supported Nextstrain environments: native, aws-batch

Setting default environment to docker.

No good.  Default environment (docker) is not supported.
```

Docker gets set as the default even though it's not supported. I think the behavior is fine since it's explicitly requested by the user, but the message `No good.  Default environment (docker) is not supported.` makes it seem like the command failed.

Suggestion: replace `No good.` with `WARNING:`. This output would read better to me:

```
# docker is not supported

...

Supported Nextstrain environments: native, aws-batch

Setting default environment to docker.

WARNING: Default environment (docker) is not supported.
```

Reply:
Yeah, I'll adjust the wording to that or something similar.

I'm also happy to replace "No good" with something else, but think `WARNING:` isn't quite the right replacement. But `ERROR:` or `Checks failed.` would be ok I think.